.github/workflows/cmake.yml

```diff
@@ -10,10 +10,22 @@ env:
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            allow-test-failure: false
+            # There sure is a lot of logic here, is there a better way?
+            should-notify-irc: ${{ github.repository == 'SerenityOS/serenity' && (github.event_name == 'pull_request' || (github.event_name == 'push' && github.ref == 'refs/heads/master')) }}
+          - os: macos-10.15
+            allow-test-failure: true
+            should-notify-irc: false
 
     steps:
     - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
```

Comment:
I don't quite see the point of this, since we don't use its main feature of selecting the exact python version. How about adding it to the list of expected-installed programs? We only ever need it on Ubuntu anyways, something like:

```diff
     - name: "Install Ubuntu dependencies"
       # These packages are already part of the ubuntu-20.04 image:
-      # cmake gcc-10 g++-10 shellcheck libgmp-dev
+      # cmake gcc-10 g++-10 python3 shellcheck libgmp-dev
```
(Package name is correct according to https://packages.ubuntu.com/focal/python3 )

Reply:
I added this because I'm also adding `flake8` to lint python, which is not part of the default ubuntu 20.04 image github gives us. Doing a `pip install` without that action makes pip install things in the wrong location.

```
Installing collected packages: mccabe, pyflakes, pycodestyle, flake8
  WARNING: The script pyflakes is installed in '/home/runner/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script pycodestyle is installed in '/home/runner/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script flake8 is installed in '/home/runner/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed flake8-3.8.4 mccabe-0.6.1 pycodestyle-2.6.0 pyflakes-2.2.0
```

Which means that the flake8 version check exits with 127 because it could not be found.,

Reply:
See this action log https://github.com/ADKaster/serenity/runs/1672776954?check_suite_focus=true

Reply:
That is to say, using the `setup-python` action ensures that a good version of python is in our $PATH, and that the version of python in our path is the one that pip installs things to. Also, apparently GitHub reserves the right to mess with what python in our path. https://github.com/actions/setup-python#using-python-without-setup-python

> The default version of Python in PATH vary between runners and can change unexpectedly so we recommend you always use setup-python.

Reply:
If we want to continue using Python, then I agree that flake8 and therefore everything Pythony is a good idea.

However, I already feel bad for having committed such a ~language crossover crime~ file, and don't think we will add more Python to Serenity anyway. C++, JS, CMake, Sh, Bash, JS, and SerenityShell is already plenty of languages.

Your choice, both options are very reasonable I think.

Reply:
> [I]  don't think we will add more Python to Serenity anyway.

What's this then? ;) https://github.com/SerenityOS/serenity/pull/4771

I  have nothing against python, it's way more approachable to me than shell scripts.

Reply:
Alrighty! Then let's go with python :D

Reply:
One can never have too many utility languages!
    - some guy that tends to make DSLs for every single problem he sees ahead of him